cuda/solver/common_trs_kernels.cuh

```diff
@@ -366,9 +368,15 @@ __global__ void sptrsv_naive_caching_kernel(
     const IndexType* const rowptrs, const IndexType* const colidxs,
     const ValueType* const vals, const ValueType* const b, size_type b_stride,
     ValueType* const x, size_type x_stride, const size_type n,
-    const size_type nrhs, bool* nan_produced)
+    const size_type nrhs, bool* nan_produced, IndexType* atomic_counter)
 {
-    const auto full_gid = thread::get_thread_id_flat<IndexType>();
+    __shared__ IndexType block_base_idx;
```

Comment:
As discussed, we need to move the `x_s_array` declaration above this one, to preserve it's alignment.